### PR TITLE
Refactor cdcmp.sh to an objdump postprocessor

### PR DESCRIPTION
**Pull request by wilzbach**
Converts the shell script into a postprocessor, s.t.:
- it works more natively with our test suite
- it can be `AUTO_UPDATE`-ed
- it can be reused (just adding `POST_SCRIPT: compilable/extra-files/objdump-postscript.sh` will enable the postprocessor for another file)
- sanitizing can be applied (https://github.com/dlang/dmd/pull/9672)

Also, auto-updating the output can be done with:

```
./run.d compilable/cdcmp.d AUTO_UPDATE=1
```

Once this is in, I will convert the other two tests too.

TODO:
- [x] investigate `objdump` differences (no `-O` was used)

### CODE REVIEW

**Comment by jacob-carlborg** [comment]
Can we use `dumpobj` instead which is shipped with the compiler. Might not run into the same issues with changed formats since it's controlled by Walter.

File: test/compilable/extra-files/objdump-postscript.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+expect_file=${EXTRA_FILES}/${TEST_NAME}.out
+obj_file=${OUTPUT_BASE}_0.o
+
+echo Creating objdump
+objdump --disassemble --disassembler-options=intel-mnemonic "${obj_file}" > "${obj_file}.dump"
```
Comment (jacob-carlborg) [inline test/compilable/extra-files/objdump-postscript.sh]: These options are not recognized on macOS.

Comment (wilzbach) [reply test/compilable/extra-files/objdump-postscript.sh]: The test isn't run on macOS (it's Linux only.)

**Comment by wilzbach** [comment]
> Can we use `dumpobj` instead which is shipped with the compiler. Might not run into the same issues with changed formats since it's controlled by Walter.

That isn't available when testing with gdc/ldc on SemaphoreCI.

File: test/compilable/extra-files/objdump-postscript.sh (hunk 1 above)
Comment (jacob-carlborg) [reply test/compilable/extra-files/objdump-postscript.sh]: Ok.

**Comment by jacob-carlborg** [comment]
> That isn't available when testing with gdc/ldc on SemaphoreCI.

Ok.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by thewilsonator** [comment]
Still fails on FreeBSD64 e.g.

```
-
  .text._D5cdcmp8test_ltzFhZb:
 
 0000000000000000 <_D5cdcmp8test_ltzFhZb>:
@@ -8,7 +7,6 @@
    4:	31 c0                	xor    eax,eax
    6:	5d                   	pop    rbp
    7:	c3                   	ret    
-
  .text._D5cdcmp8test_lezFhZb:
 
 0000000000000000 <_D5cdcmp8test_lezFhZb>:
@@ -22,7 +20,6 @@
   15:	5d                   	pop    rbp
   16:	c3                   	ret    
 	...
-
  .text._D5cdcmp8test_eqzFhZb:
 
 0000000000000000 <_D5cdcmp8test_eqzFhZb>:
@@ -36,7 +33,6 @@
   15:	5d                   	pop    rbp
   16:	c3                   	ret    
 	...
-
  .text._D5cdcmp8test_nezFhZb:
 
 0000000000000000 <_D5cdcmp8test_nezFhZb>:
@@ -50,7 +46,6 @@
   15:	5d                   	pop    rbp
   16:	c3                   	ret    
 	...
-
```

**Comment by WalterBright** [comment]
https://github.com/dlang/dmd/pull/9705 seems to have a much simpler solution.

**Comment by wilzbach** [comment]
#9705 doesn't allow auto-updating nor reusing.

File: test/compilable/extra-files/objdump-postscript.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+echo SANITIZING Objdump...
+< "${obj_file}.dump" \
+    tail -n+3 | \
+    sed s/^.*section// > "${obj_file}.dump.sanitized"
```
Comment (WalterBright) [inline test/compilable/extra-files/objdump-postscript.sh]: don't need the sed script anymore

Comment (wilzbach) [reply test/compilable/extra-files/objdump-postscript.sh]: Yes, we do - to support FreeBSD.

Comment (wilzbach) [reply test/compilable/extra-files/objdump-postscript.sh]: Removed it as the current solution doesn't support FreeBSD either :O

**Comment by wilzbach** [comment]
Should be passing now.

**Comment by Basile-z** [comment]
Sorry but the "localized messages" issue is back.

**Pull request merged**